File: src/main/java/com/rwbymod/item/ItemRegistry.java
package com.rwbymod.item;

import com.rwbymod.misc.EnumDustStimulus;

import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fml.common.registry.GameRegistry;
import net.minecraftforge.oredict.OreDictionary;

public class ItemRegistry {

	public static Item dust;
	public static Item dustCrystal;
	public static Item scroll;
	
	public static void init() {
		dust = register(new ItemDust());
		dustCrystal = register(new ItemDustContainer("dust_crystal") {});
		scroll = register(new ItemScroll());
	}
	
	// ----- UTILITY FUNCTIONS ----- //
	
	private static Item register(Item item) { return GameRegistry.register(item); }
	
}

File: src/main/scala/com/rwbymod/registry/ItemRegistry.scala
package com.rwbymod.registry

import com.rwbymod.item.ItemDust
import com.rwbymod.item.ItemDustContainer
import com.rwbymod.item.ItemScroll

import net.minecraft.item.Item
import net.minecraftforge.fml.common.registry.GameRegistry

object ItemRegistry {

  val dust = new ItemDust()
  val dustCrystal = new ItemDustContainer("dust_crystal"){}
  val scroll = new ItemScroll()

  def init {
    Array[Item](
      dust,
      dustCrystal,
      scroll
    ).foreach { GameRegistry.register[Item] }
  }

}